client/blocks/importer/wordpress/upgrade-plan/index.tsx

```diff
@@ -54,7 +54,20 @@ export const UpgradePlan: React.FunctionComponent< Props > = ( props: Props ) =>
 		// If the user's email is unverified, we still want to show the trial plan option
 		migrationTrialEligibility?.error_code === 'email-unverified';
 
+	const plans = Plans.useSitePlans( { siteId: site.ID } );
+	const pricing = plans.data ? plans.data[ visiblePlan ].pricing : undefined;
+
+	const introOfferAvailable =
+		isEnabled( 'migration-flow/introductory-offer' ) &&
+		pricing?.introOffer &&
+		pricing.introOffer.rawPrice &&
+		! pricing.introOffer.isOfferComplete &&
+		pricing.originalPrice.monthly &&
+		pricing.originalPrice.full &&
+		pricing.currencyCode;
+
 	const hideFreeMigrationTrial =
+		isEnabled( 'migration-flow/introductory-offer' ) ||
```

Comment:
Minor thought: much like monthly plans, this will hide the trial from all users, even those not eligible for the introductory offer. Maybe we should just disable the free trial when we have an introductory offer available.
```suggestion
		introOfferAvailable ||
```